lib/Sema/ConstraintLocator.cpp

```diff
@@ -249,6 +250,11 @@ bool ConstraintLocator::isForOptionalTry() const {
   return directlyAt<OptionalTryExpr>();
 }
 
+bool ConstraintLocator::isForFunctionBuilderBodyResult() const {
+  auto elt = getFirstElementAs<LocatorPathElt::FunctionBuilderBodyResult>();
+  return elt.hasValue();
```

Comment:
Nit: `ConstraintLocator` has `isLastElement` helper which could be used here.

Reply:
There's no equivalent `isFirstElement` - should I add one?

Reply:
Yeah, I think that would be useful!

Reply:
✅